### PR TITLE
Raise Error when single sequence is passed to motifs.Instances

### LINKED ISSUE #971: Bio.Motifs throws KeyError with IUPAC.ambiguous_dna
Hi all,

I'm trying to create a couple of test Seq objects to do a comparison between them as proof of concept for running it on a big set of Seq objects using PySpark and map.  However, when I create a Seq object with an N in it, even if I use IUPAC.ambiguous_dna, it throws a KeyError on the N.  Should I be using a different alphabet or doing something else different?

```
Python 3.5.0 (default, Sep 25 2015, 12:42:09)
[GCC 4.4.7 20120313 (Red Hat 4.4.7-11)] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> from Bio.Alphabet import IUPAC
>>> from Bio.Seq import Seq
>>> from Bio import motifs
>>> t1 = Seq("ATCGGN", IUPAC.ambiguous_dna)
>>> t1
Seq('ATCGGN', IUPACAmbiguousDNA())
>>> motifs.create(t1)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/tools/lib/python3.5/site-packages/Bio/motifs/__init__.py", line 25, in create
    return Motif(instances=instances, alphabet=alphabet)
  File "/tools/lib/python3.5/site-packages/Bio/motifs/__init__.py", line 239, in __init__
    counts = self.instances.count()
  File "/tools/lib/python3.5/site-packages/Bio/motifs/__init__.py", line 197, in count
    counts[letter][position] += 1
KeyError: 'N'
```

### PR DESCRIPTION
**Pull request by Caiofcas**
- [X] I hereby agree to dual licence this and any previous contributions under both
the _Biopython License Agreement_ **AND** the _BSD 3-Clause License_.

- [X] I have read the ``CONTRIBUTING.rst`` file, have run ``pre-commit`` locally,
and understand that AppVeyor and TravisCI will be used to confirm the Biopython unit
tests and style checks pass with these changes.

- [X] I have added my name to the alphabetical contributors listings in the files
``NEWS.rst`` and ``CONTRIB.rst`` as part of this pull request, am listed
already, or do not wish to be listed. (*This acknowledgement is optional.*)

Closes #971

### CODE REVIEW

File: Bio/motifs/__init__.py [hunk 1]
```diff
@@ -20,9 +20,19 @@
...
 
 def create(instances, alphabet="ACGT"):
     """Create a Motif object."""
+    if isinstance(instances, Seq) or isinstance(instances, str):
+        warnings.warn(
+            "instances should be iterator of Seq objects or strings."
+            "If a single sequence is given, will treat each character as a "
+            "separate sequence.",
+            BiopythonWarning,
+        )
```
Comment (mdehoon) [inline Bio/motifs/__init__.py:35]: Shouldn't this be in `Instances.__init__`?

Comment (mdehoon) [reply Bio/motifs/__init__.py:35]: Also, `Seq` and `str` can be checked both with a single call to `isinstance`. We should also include `MutableSeq`.

Comment (Caiofcas) [reply Bio/motifs/__init__.py:35]: You are right in both, I also realized I misread the issue and it's supposed to raise an error and not a warning.

**Comment by mdehoon** [comment]
Thanks!

**Pull request merged**